website/content/docs/upgrading/identity-deduplication.mdx

```diff
@@ -0,0 +1,406 @@
+   so, how to safely resolve them.
+---
+
+# Vault identity duplicate resolution guide
+
+Some users may have duplicate identity resources (entities, aliases, groups) in
+their cluster's storage due to historical bugs. These can cause unexpected
+behavior as they are outside of Vault's typical expectations and test scenarios.
+
+Vault 1.19.0 includes additional reporting in server logs to identify if any
+duplicates exist in a cluster. If any do exist, this document provides the
+information necessary to safely resolve them and return the cluster to normal,
+supported behavior.
+
+## Diagnosis
+
+To identify whether your cluster has duplicates you need to follow these steps:
+
+ 1. Ensure you have upgraded to Vault 1.19.0 or higher on all cluster nodes.
+ 2. Check the Vault logs on the active node and locate it's last unseal operation between the following log lines:
+    ```
+    [INFO]  core: post-unseal setup starting
+    ...<check here> ...
+    [INFO]  core: post-unseal setup complete
+    ```
+ 3. If you see the following line, duplicates were found and you should continue
+    to read this guide.
+
+    ```
+    [WARN]  identity: DUPLICATES DETECTED, see following logs for details [...]
+    ```
+
+    **If you don't see this line, you have no further action to take on this
+    cluster**. Ensure you repeat this process on any Performance Replication
+    (PR) secondary clusters though as it's possible for PR secondaries to have
+    local duplicates. Disaster Recovery (DR) secondaries do not need to be
+    checked separately.
+
+## Resolution
+
+If duplicates have been reported in your Vault logs, there are a few steps to
+work through which will depend on the types of duplicates found. Details of each
+are given below. The high-level process for resolution is:
+
+#### With Vault Enterprise Performance or Disaster Recovery replication
+
+ 1. Ensure all clusters (primary and secondaries) are upgraded to 1.19.0 or
+    higher.
+ 1. On the primary, address each duplicate reported using the type-specific
+    guidance below.
+ 2. On any Performance Replication secondaries, address each _local alias_ duplicate reported using the guidance below.
+ 3. Activate the `force-identity-deduplication` activation flag.
+
+#### Without Vault replication
+
+ 1. Ensure the cluster is upgraded to 1.19.0 or higher.
+ 1. Address each duplicate reported using the type specific guidance below.
+ 2. Activate the `force-identity-deduplication` activation flag.
+
+### Resolving different-case entity alias duplicates
+
+```
+[WARN]  identity: 2 different-case entity alias duplicates found (potential security risk)
+[WARN]  identity: entity-alias "alias-case" with mount accessor "auth_userpass_34aca7ec" duplicates 1 others: id=df3568a4-3b65-4104-9481-1129ecbed72f canonical_id=5f013d99-a6c7-9a00-6ad5-4ad724b14f60 force_deduplication="would merge into entity 7da76b0d-fe9b-a125-3362-2a8ff055dcf8"
```

Comment:
Just a question: Feel free to take this offline as it's not related to this PR and probably more to my overall noobery, but I'm just curious what a canonical ID is vs the regular ID. In this example log line I was first trying to find the corresponding entity 7da7 and didn't realize it was the canonical_id I was supposed to be looking at, not the id.

Is there a way for users to query identity resources by canonical ID to inspect them, and see "aha, they're right, it's merged"? Or can you only query by id? I'm just curious from a UX perspective.